### PR TITLE
Add Avro file format product tests

### PR DESCRIPTION
**Pull request by anusudarsan**
Known limitations:
AVRO file format doesn't support tinyint and smallint datatypes.
AVRO file format doesn't support coercion from bigint to string datatype.

### CODE REVIEW

**Comment by anusudarsan** [comment]
supersedes https://github.com/prestodb/presto/pull/8099

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java [hunk 1]
```diff
@@ -211,6 +224,63 @@ public void testSelectAllDatatypesRcfile()
...
+                "346," +
+                "345.67800," +
+                "'" + parseTimestampInLocalTime("2015-05-10 12:15:35.123", PRODUCT_TESTS_TIME_ZONE).toString() + "'," +
+                "'" + Date.valueOf("2015-05-10") + "'," +
+                "'ala ma kota'," +
+                "'ala ma kot'," +
+                "'ala ma    '," +
+                "true," +
+                "'kot binarny'" +
+                ")", tableName));
```
Comment (kokosing) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java]: put `tableName` into the next line

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java [hunk 2]
```diff
@@ -211,6 +224,63 @@ public void testSelectAllDatatypesRcfile()
...
+                row("c_decimal_w_params", "decimal(10,5)"),
+                row("c_timestamp", "timestamp"),
+                row("c_date", "date"),
+                row("c_string", "varchar"),
+                row("c_varchar", "varchar(10)"),
+                row("c_char", "char(10)"),
+                row("c_boolean", "boolean"),
+                row("c_binary", "varbinary"));
+
+        QueryResult queryResult = query(format("SELECT * FROM %s", tableName));
```
Comment (kokosing) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java]: no need to use `format`

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java [hunk 3]
```diff
@@ -211,6 +224,63 @@ public void testSelectAllDatatypesRcfile()
...
+                "'" + parseTimestampInLocalTime("2015-05-10 12:15:35.123", PRODUCT_TESTS_TIME_ZONE).toString() + "'," +
+                "'" + Date.valueOf("2015-05-10") + "'," +
+                "'ala ma kota'," +
+                "'ala ma kot'," +
+                "'ala ma    '," +
+                "true," +
+                "'kot binarny'" +
+                ")", tableName));
+
+        assertThat(query(format("SHOW COLUMNS FROM %s", tableName)).project(1, 2)).containsExactly(
```
Comment (kokosing) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java]: no need to use `format`

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java [hunk 4]
```diff
@@ -275,6 +345,46 @@ else if (usingTeradataJdbcDriver(connection)) {
         }
     }
 
+    private void avroAssertColumnTypes(QueryResult queryResult)
```
Comment (kokosing) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java]: please inline this method. It matches only `QueryResult` for particular query so it cannot be reused.

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java [hunk 5]
```diff
@@ -275,6 +345,46 @@ else if (usingTeradataJdbcDriver(connection)) {
...
+                    DECIMAL,
+                    TIMESTAMP,
+                    DATE,
+                    LONGNVARCHAR,
+                    LONGNVARCHAR,
+                    CHAR,
+                    BOOLEAN,
+                    LONGVARBINARY);
+        }
+        else if (usingTeradataJdbcDriver(connection)) {
```
Comment (kokosing) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java]: have you tested that with simba drivers?

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java [hunk 6]
```diff
@@ -115,6 +119,18 @@
                         "STORED AS PARQUET");
     }
 
+    private static HiveTableDefinition.HiveTableDefinitionBuilder avroTableDefinitionBuilder()
```
Comment (kokosing) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java]: this method is used only once, so you do not need to return builder.

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java [hunk 7]
```diff
@@ -205,6 +231,37 @@ public void testHiveCoercionParquet()
...
+                        "(-2323, -1.5)",
+                tableName));
+
+        avroAlterTableColumnTypes(tableName);
+        avroAssertProperAlteredTableSchema(tableName);
+
+        QueryResult queryResult = query(format("SELECT * FROM %s", tableName));
+        avroAssertColumnTypes(queryResult);
+        assertThat(queryResult).containsOnly(
+                row(
```
Comment (kokosing) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java]: make `row(...)` a single line

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java [hunk 8]
```diff
@@ -291,6 +356,20 @@ else if (usingTeradataJdbcDriver(connection)) {
         }
     }
 
+    private void avroAssertColumnTypes(QueryResult queryResult)
+    {
+        Connection connection = defaultQueryExecutor().getConnection();
+        if (usingPrestoJdbcDriver(connection) || usingTeradataJdbcDriver(connection)) {
```
Comment (kokosing) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java]: so do we need this if?

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java [hunk 9]
```diff
@@ -303,6 +382,12 @@ private static void alterTableColumnTypes(String tableName)
         executeHiveQuery(format("ALTER TABLE %s CHANGE COLUMN float_to_double float_to_double double", tableName));
     }
 
+    private static void avroAlterTableColumnTypes(String tableName)
```
Comment (kokosing) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveCoercion.java]: I would inline this method, and all avro related ones in this file

File: presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java (hunk 5 above)
Comment (anusudarsan) [reply presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestAllDatatypesFromHiveConnector.java]: not yet. I added it to simba_jdbc group. will test it

**Comment by wenleix** [comment]
cc @wagnermarkd  you might be interested in this PR

File: presto-product-tests/src/main/java/com/facebook/presto/tests/JmxConnectorTests.java [hunk 10]
```diff
@@ -31,7 +32,7 @@
 public class JmxConnectorTests
         extends ProductTest
 {
-    @Test(groups = JMX_CONNECTOR)
+    @Test(groups = {JMX_CONNECTOR, SIMBA_JDBC})
```
Comment (kokosing) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/JmxConnectorTests.java]: maybe put it into just `jdbc`, it tests also public jdbc. Please change it for other tests as well.

**Comment by anusudarsan** [comment]
updated. tested using simba driver as well.

**Comment by kokosing** [comment]
Can you additionally add tests which is falling on CDH to SKIP_ON_CDH test group?

**Comment by anusudarsan** [comment]
done

**Pull request merged**